src/process_request.cpp

```diff
@@ -34,10 +34,18 @@ void validate_user_db_update_permission (
   if (!user_id)
     throw http::unauthorized ("User is not authorized");
 
-  if (selection.supports_user_details ()
-      && selection.is_user_blocked (*user_id))
-    throw http::forbidden (
-	"Your access to the API has been blocked. Please log-in to the web interface to find out more.");
+  if (selection.supports_user_details ())
+  {
+    if (selection.is_user_blocked (*user_id))
+      throw http::forbidden (
+	  "Your access to the API has been blocked. Please log-in to the web interface to find out more.");
+
+    // User status has to be either active or confirmed, otherwise the request
+    // to change the database will be rejected
+    if (!selection.is_user_active(*user_id))
+      throw http::forbidden (
+          "You have not permitted the application access to this facility");
```

Comment:
That's not really an accurate description is it? This would occur when there was an authorization but the user account had been closed or suspended.

Reply:
That’s exactly what Rails returned in that case, I’m only trying to replicate the same response here. I agree, the description isn’t all that helpful.

Reply:
api_controller.rb deny_access is where this error message originates from:

```ruby
  def deny_access(_exception)
    if doorkeeper_token || current_token
      set_locale
      report_error t("oauth.permissions.missing"), :forbidden
```